### PR TITLE
Added development docker compose

### PR DESCRIPTION
**Pull request by omn0mn0m**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [submodule "assets/css"]
 	path = assets/css
-	url = git@github.com:Glimesh/glimesh.tv-styles.git
+	url = ../../Glimesh/glimesh.tv-styles.git
```
Comment (clone1018) [inline .gitmodules:3]: Is this right?

Comment (omn0mn0m) [reply .gitmodules:3]: I would try it, but it worked for me in WSL 2 and in Manjaro.

Comment (omn0mn0m) [reply .gitmodules:3]: Confirmed by cloning a new repository and trying to spin up. Also found a StackOverflow question confirming my solution: https://stackoverflow.com/questions/40841882/automatically-access-git-submodules-via-ssh-or-https

Comment (clone1018) [reply .gitmodules:3]: Oh interesting, so it back two directories on GitHub... Not on your local filesystem.

File: Dockerfile.dev [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+RUN mix local.hex --force
+RUN mix local.rebar --force
+
+# install node
+RUN curl -sL https://deb.nodesource.com/setup_12.x | bash -
+RUN apt-get install -y nodejs
+
+# prepare build dir
+WORKDIR /app
+EXPOSE 4000
```
Comment (haydenmc) [inline Dockerfile.dev:17]: Also `4001` for HTTPS?

Comment (omn0mn0m) [reply Dockerfile.dev]: Done :)

File: run.sh [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+set -e
+
+# install dependencies
+mix deps.get
+
+# create and migrate database
+mix ecto.setup
+
+# install nodejs stuff
+cd assets && npm install
```
Comment (clone1018) [inline run.sh:12]:
```suggestion
cd assets && npm ci
```

Can you reset the package lock, and change this script to use ci? That we we don't accidentally upgrade the whole repo?

**Pull request merged**